docs/config.md

```diff
@@ -1357,7 +1368,7 @@ Below is a list of the most common gateway setups.
        }
      }'
    ```
-   - **Performance:** consider running with `Routing.AcceleratedDHTClient=true` and either `Provider.Enabled=false` (avoid providing newly retrieved blocks) or `Provider.WorkerCount=0` (provide as fast as possible, at the cost of increased load)
+   - **Performance:** Consider enabling `Routing.AcceleratedDHTClient=true` to improve content routing lookups. Separately, gateway operators should decide if the gateway node should also co-host and provide (announce) fetched content to the DHT. If providing content, enable `Provide.DHT.SweepEnabled=true` for efficient announcements. If announcements are still not fast enough, adjust `Provide.DHT.MaxWorkers`. For a read-only gateway that doesn't announce content, use `Provide.Enabled=false`.
```

Comment:
This option living in Routing, with the actual provider system options living in Provide feels weird.

Also, do we want to keep supporting a world where we do very innefficient reprovides? One option is to migrate as follows:

* AcceleratedDHTClient = true -> Legacy Reprovider enabled (no behaviour change)
* AcceleratedDHTClient = false -> Sweeping reproviding enabled (class upgrade)
* And remove the AcceleratedDHTClient option itself.

Reply:
We want to enable Sweeping Provider by default in v0.39.

From the benchmarks, the Sweeping Provider performance is approx the same (slightly higher), than Legacy System with AcceleratedDHTClient.

Note that it is possible to combine Sweeping Provider with AcceleratedDHTClient (it won't change the provide performance much, and it will still have the spikes when crawling the DHT swarm). This can be helpful if users need very quick lookups (FIND_NODE, FIND_PROVS), but it won't change much for providing.

Ideally, the Provide config parameters that are specific to a DHT Swarm (all but `Provide.Enabled` and `Provide.Strategy`) should move to the appropriate `Routing.Routers` component (that will be a DHT router), and `AcceleratedDHTClient` should move there too.